test/common/secret/secret_manager_impl_test.cc

```diff
@@ -199,6 +217,284 @@ name: "abc.com"
+  last_updated:
+    seconds: 1234567891
+    nanos: 234000000
+  secret:
+    name: "abc.com"
+  )EOF";
+  checkConfigDump(expected_secrets_config_dump);
+
+  time_system_.setSystemTime(std::chrono::milliseconds(1234567899000));
+  auto context_secret_provider = secret_manager->findOrCreateCertificateValidationContextProvider(
+      config_source, "abc.com.validation", secret_context);
+  init_target_handle->initialize(init_watcher);
+  const std::string updated_config_dump = R"EOF(
+dynamic_warming_secrets:
+- name: "abc.com"
+  version_info: "uninitialized"
+  last_updated:
+    seconds: 1234567891
+    nanos: 234000000
+  secret:
+    name: "abc.com"
+- name: "abc.com.validation"
+  version_info: "uninitialized"
+  last_updated:
+    seconds: 1234567899
+  secret:
+    name: "abc.com.validation"
+)EOF";
+  checkConfigDump(updated_config_dump);
+}
+
+TEST_F(SecretManagerImplTest, ConfigDumpHandlerStaticSecrets) {
+  Server::MockInstance server;
+  auto secret_manager = std::make_unique<SecretManagerImpl>(config_tracker_);
+  time_system_.setSystemTime(std::chrono::milliseconds(1234567891234));
+
+  NiceMock<Server::Configuration::MockTransportSocketFactoryContext> secret_context;
+
+  envoy::api::v2::core::ConfigSource config_source;
+  NiceMock<LocalInfo::MockLocalInfo> local_info;
+  NiceMock<Event::MockDispatcher> dispatcher;
+  NiceMock<Runtime::MockRandomGenerator> random;
+  Stats::IsolatedStoreImpl stats;
+  NiceMock<Init::MockManager> init_manager;
+  NiceMock<Init::ExpectableWatcherImpl> init_watcher;
+  Init::TargetHandlePtr init_target_handle;
+  EXPECT_CALL(init_manager, add(_))
+      .WillRepeatedly(Invoke([&init_target_handle](const Init::Target& target) {
+        init_target_handle = target.createHandle("test");
+      }));
+  EXPECT_CALL(secret_context, stats()).WillRepeatedly(ReturnRef(stats));
+  EXPECT_CALL(secret_context, initManager()).WillRepeatedly(Return(&init_manager));
+  EXPECT_CALL(secret_context, dispatcher()).WillRepeatedly(ReturnRef(dispatcher));
+  EXPECT_CALL(secret_context, localInfo()).WillRepeatedly(ReturnRef(local_info));
+
+  const std::string tls_certificate =
+      R"EOF(
+name: "abc.com"
+tls_certificate:
+  certificate_chain:
+    inline_string: "DUMMY_INLINE_BYTES_FOR_CERT_CHAIN"
+  private_key:
+    inline_string: "DUMMY_INLINE_BYTES_FOR_PRIVATE_KEY"
+  password:
+    inline_string: "DUMMY_PASSWORD"
+)EOF";
+  envoy::api::v2::auth::Secret tls_cert_secret;
+  TestUtility::loadFromYaml(TestEnvironment::substitute(tls_certificate), tls_cert_secret);
+  secret_manager->addStaticSecret(tls_cert_secret);
+  const std::string expected_config_dump = R"EOF(
+static_secrets:
+- name: "abc.com" 
+  secret:
+    name: "abc.com"
+    tls_certificate:
+      certificate_chain:
+        inline_string: "DUMMY_INLINE_BYTES_FOR_CERT_CHAIN"
+      private_key:
+        inline_string: "[redacted]"
+      password:
+        inline_string: "[redacted]"
+)EOF";
+  checkConfigDump(expected_config_dump);
```

Comment:
Could you add one more certificate here, without password, so that we have coverage for when sensitive information is set and sanitized, and when there is no sensitive information (to verify that it's not always set to `"[redacted]"`)? I know it's handled correctly, but I don't see that verify it.

Reply:
Done. Added that case.